src/Form/index.tsx

```diff
@@ -1353,6 +1403,13 @@ function Form({
       });
     } else if (link === LINK_BACK) {
       await goToPreviousStep();
+    } else if (link === LINK_COLLECT_PAYMENT) {
+      await collectPaymentAction(button);
+      // For custom payment, also submit and navigate
+      await buttonOnSubmit({
```

Comment:
does submission / navigation always need to happen? couldn't collectPayment just open up a new tab with the stripe checkout session and leave the current step as is?

Reply:
There are two issues to consider in your comment.  

The first is the part about opening the stripe checkout in a new tab.  We certainly could do this but should we?  Also, would we always open the checkout page in a new tab or would it be a form designer's choice?  If it is a choice, then we need another property for it.  In my opinion, opening in another tab is unlikely to be what most people will want and I suggest that we leave this feature out until someone needs it.

With the above in mind, it was my thinking that with stripe checkout there is always a synchronous redirect happening in collectPaymentAction before the buttonOnSubmit call here.  Basically, the buttonOnSubmit call only ever happens with a custom payment.  This is why I coded it this way.

Reply:
Per our discussion, leaving this as is and not supporting checkout in another tab/window for rn.